docs/src/spec/to_mem_function.md

```diff
@@ -0,0 +1,9 @@
+# The `to_mem` function
+
+Reference type values can be copied from storage and into memory using the `to_mem` function.
```

Comment:
In the future, I think there may be somewhat nuanced usage of  `to_mem()` on values in memory.

Example:

```
pub fn foo(self):
  sum(self.my_array)
  sum([1, 2, 3]) # this is not copied in `sum`, as `to_mem` operates on it passively

fn sum(some_array: u8[3]):
  let sum: u256 = 0
  for val in some_array.to_mem():
    sum += val

  return sum
```

So I think we should remove the check mandating that the value be in storage (if it exists). This would of course be a separate issue.

Reply:
But why would we even write `some_array.to_mem()` here? Isn't the assumption that `some_array` is in memory anyway? Isn't it that we would only need to call `to_mem()` on something that is `self.<something>`? Or is this hypothetical code in which `some_array` may allow *readonly* storage pointers?

Also we might want to reconsider if `to_mem` can be merged with `clone`.

Reply:
> Isn't the assumption that some_array is in memory anyway?

This is the current assumption, as we require that reference type parameters be in memory.

> Or is this hypothetical code in which some_array may allow readonly storage pointers?

This is what has been described in #161. `to_mem` would serve as a function that would guarantee that a readonly value is in memory. If the value is already in memory, then we would leave it alone, as to not incur unnecessary gas costs. If the value is in storage, then we would copy it.